src/contour/opengl/TerminalWidget.cpp

```diff
@@ -491,17 +497,11 @@ void TerminalWidget::initializeGL()
     configureScreenHooks();
     watchKdeDpiSetting();
 
-    renderer_.initialize(terminal().pageSize(),
-                         sanitizeFontDescription(profile().fonts, fontDPI()),
-                         terminal().screen().colorPalette(),
-                         profile().backgroundOpacity,
-                         session_.config().textureAtlasHashtableSlots,
-                         session_.config().textureAtlasTileCount,
-                         session_.config().textureAtlasDirectMapping,
-                         profile().hyperlinkDecoration.normal,
-                         profile().hyperlinkDecoration.hover
-                         // TODO: , WindowMargin(windowMargin_.left, windowMargin_.bottom);
-    );
+#if defined(CONTOUR_INIT_RENDERER_EARLY)
+    applyFontDPI();
+#else
+    initializeRenderer();
+#endif
```

Comment:
What is the problem with spawning the renderer early in every case? This macro magic is ***really*** weird. Also, why `#if defined` instead of `#ifdef`?

Reply:
I prefer `#if defined` over `#ifdef` as it's more expressive. Apart from that, nothing else. :-)